### PR TITLE
Add support for nodegroup matching and update to newer deps

### PR DESCRIPTION
**Pull request by huxcrux**
We need to support upgrading from v1alpha1 however this is working asis if the old CRD is removed first.

You are able to update from the latest release v0.3 directly to this version. However will need to update your resource from `apiVersion: k8s.elx.cloud/v1alpha1` to `apiVersion: k8s.elx.cloud/v1alpha2` in order to use the new nodegroup matching

### CODE REVIEW

File: hux-lab.yaml
Comment (zrk02) [inline hux-lab.yaml:1]: Is this file intended to be included or did it just happen to be included?

Comment (huxcrux) [reply hux-lab.yaml:1]: It was use for testing and not intended to be a part of this PR.
I removed the file again :)

**Review by zrk02 — APPROVED** [review]
A lot going on here. 
Cannot say more than it seems reasonable and at least not blatantly fishy.
